### PR TITLE
Fixed saving of best_weights

### PR DESCRIPTION
**Pull request by jesperdlau**
Hi there. I noticed a small bug in the way DeepDTA is retrained. 
From [this pytorch tutorial](https://pytorch.org/tutorials/beginner/saving_loading_models.html): 
_"If you only plan to keep the best performing model (according to the acquired validation loss), don’t forget that best_model_state = model.state_dict() returns a reference to the state and not its copy! You must serialize best_model_state or use best_model_state = deepcopy(model.state_dict()) otherwise your best best_model_state will keep getting updated by the subsequent training iterations. As a result, the final model state will be the state of the overfitted model."_

Our retraining results in the following RMSE for DeepDTA:
Train: 1.38
Validation: 1.43
Test: 1.68
BDB2020+: 1.71

But you will have to redo the training yourselves, just to be sure.

### CODE REVIEW

**Comment by KSUN63** [comment]
Thanks for pointing it out! we will also update our training result etc as well
